### PR TITLE
Connect Frontend with API for mentorship programme overview 

### PR DESCRIPTION
**Pull request by joanaBrit**
## Description

I implemented the connection between the frontend and the API for the mentorship program overview page. 

This change is necessary to ensure that the mentorship program overview page displays the data from the backend.

Fixes issue #26

## Type

- [ ] Bug Fix
- [✅] New Feature
- [ ] Code Refactor
- [ ] Documentation
- [ ] Other

## Screenshots

Before:
Screenshot 2024-09-10 123455

After:
Screenshot 2024-09-10 121847

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [✅] I checked and followed the [contributor guide](../CONTRIBUTING.md)
- [✅] I have tested my changes locally.
- [✅] I have added a screenshot from the website after I tested it locally

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ interface HomePageProps {
 }
 
 const HomePage = ({ data, footer, error }: HomePageProps) => {
+  //
```
Comment (stepsen89) [inline src/pages/index.tsx:26]: You can delete that

File: src/pages/mentorship/index.tsx [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
-      <Typography variant="h4">Welcome to the Mentorship Page</Typography>
-    </div>
-  );
+const MentorshipPage = ({ mentorship }: MentorshipPageProps) => {
+  return <div>{JSON.stringify(mentorship)}</div>;
+};
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  try {
+    const combinedResponse = await fetchData('mentorship/overview');
```
Comment (stepsen89) [inline src/pages/mentorship/index.tsx:13]: Can you create a type here? Would be a new type for the mentorship overview response and the footer (combined)

File: src/pages/mentorship/index.tsx [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+
+    return {
+      props: {
+        mentorship: combinedResponse.data,
```
Comment (stepsen89) [inline src/pages/mentorship/index.tsx:17]: Not sure about the design, but is there a footer here - the footer can be added the same way as in the index page

File: src/pages/mentorship/index.tsx [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
+import { MentorshipPageProps } from '@utils/types';
+import { fetchData } from 'lib/api';
 
-const MentorshipPage = () => {
-  return (
-    <div>
-      <Typography variant="h4">Welcome to the Mentorship Page</Typography>
-    </div>
-  );
+const MentorshipPage = ({ mentorship }: MentorshipPageProps) => {
```
Comment (stepsen89) [inline src/pages/mentorship/index.tsx:7]: PageProps should be in the page, it belongs here and won't be needed in other pages

File: src/utils/types.ts [hunk 5]
```diff
@@ -74,3 +74,44 @@ export type FooterResponse = {
...
+};
+
+export type MentorshipProgrammeData = {
+  page: PageProps;
+  mentorSection: MentorOrMenteeProps;
+  menteeSection: MentorOrMenteeProps;
+  feedback: FeedbackProps;
+};
+
+export type PageProps = {
```
Comment (stepsen89) [inline src/utils/types.ts:90]: Could you explain what this is supposed to be? PageProps is misleading

File: src/utils/types.ts [hunk 6]
```diff
@@ -74,3 +74,44 @@ export type FooterResponse = {
...
+  description: string;
+};
+
+export type LinkProps = {
+  label: string;
+  url: string;
+};
+
+export type MentorOrMenteeProps = {
+  title: string;
```
Comment (stepsen89) [inline src/utils/types.ts:101]: Types are usually not called Props - just call it something meaningful so it's clear what it is 

Do we distinguish Mentor and Mentee at some point?

File: src/pages/mentorship/index.tsx (hunk 3 above)
Comment (joanaBrit) [reply src/pages/mentorship/index.tsx:17]: In the design doesn't appear any footer, how do I know that I need to include footer? Yes, I can add the footer.

File: src/utils/types.ts (hunk 6 above)
Comment (joanaBrit) [reply src/utils/types.ts:101]: Maybe we can change the this type when we do distinguish them in the future? For now they are the same in the API response. I've renamed these types in the mean time

File: src/pages/mentorship/index.tsx (hunk 3 above)
Comment (stepsen89) [reply src/pages/mentorship/index.tsx:22]: No need to add the footer if it's not in the design then

**Review by stepsen89 — APPROVED** [review]

**Pull request merged**